modules/ding_gatewayf/modules/ding_gatewayf_registration/ding_gatewayf_registration.module

```diff
@@ -0,0 +1,456 @@
+    '#collapsible' => TRUE,
+    '#collapsed' => TRUE,
+    '#tree' => TRUE,
+  );
+
+  $form['ding_gatewayf_registration']['acceptance']['description'] = array(
+    '#type' => 'textarea',
+    '#title' => t('Acceptance text'),
+    '#description' => t('Description displayed next to the user acceptance check box. This field accept HTML'),
+    '#default_value' => $defaults['acceptance']['description'],
+  );
+
+  // Define after create info page.
+  $form['ding_gatewayf_registration']['success'] = array(
+    '#type' => 'fieldset',
+    '#title' => t('Successful registration page'),
+    '#description' => t('Information displayed after the user have been successful registered at the provider.'),
+    '#tree' => TRUE,
+    '#collapsible' => TRUE,
+    '#collapsed' => TRUE,
+  );
+
+  $form['ding_gatewayf_registration']['success']['title'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Title'),
+    '#default_value' => $defaults['success']['title'],
+  );
+
+  $form['ding_gatewayf_registration']['success']['content'] = array(
+    '#type' => 'textarea',
+    '#title' => t('Content'),
+    '#description' => t('This field accept HTML.'),
+    '#default_value' => $defaults['success']['content'],
+  );
+}
+
+/**
+ * Generate render array for the information page.
+ *
+ * @return array
+ *   Render array with the page content.
+ */
+function ding_gatewayf_registration_information() {
+  $config = _ding_gatewayf_registration_load_config();
+
+  // Set page title to the same as the pages headline.
+  drupal_set_title($config['information']['title']);
+
+  return array(
+    '#theme' => 'ding_gatewayf_registration_information',
+    '#title' => $config['information']['title'],
+    '#content' => $config['information']['content'],
+    '#link' => array(
+      '#theme' => 'link',
+      '#text' => $config['information']['link'],
+      '#path' => DING_GATEWAYF_LOGIN_URL,
+      '#options' => array(
+        'attributes' => array(
+          // This class "form-submit" is used to make the link look and behave
+          // link a button.
+          'class' => array(
+            'form-submit',
+            'ding-gatewayf-link',
+          ),
+        ),
+        'query' => array(
+          'destination' => DING_GATEWAYF_REGISTRATION_ACCEPTANCE_URL,
+        ),
+        'html' => TRUE,
+      ),
+    ),
+  );
+}
+
+/**
+ * Implements hook_form().
+ */
+function ding_gatewayf_registration_acceptance_form() {
+  $config = _ding_gatewayf_registration_load_config();
+
+  // This is a little style hack to get around the panels design requirements of
+  // the DDB default theme. The module existing_pages could have been used to
+  // wrapper the form in panels. But seen form a performance point this is
+  // faster and simpler.
+  $form = array(
+    '#prefix' => '<div class="empty-sidebars default-layout"><div class="layout-wrapper"><div class="pane-content"><div class="primary-content">',
+    '#suffix' => '</div></div></div></div>',
+  );
```

Comment:
I image this could be a problem that might occour multiple times. Could we extract it into a theme function and use [`#theme_wrappers`](https://api.drupal.org/api/drupal/developer%21topics%21forms_api_reference.html/7#theme_wrappers)?